desk/app/channels.hoon

```diff
@@ -1213,38 +1209,39 @@
     :-  recency.remark.channel
     =/  unreads
       %+  skim
-        %~  tap  by
+        %-  bap:on-v-posts:c
         (lot:on-v-posts:c posts.channel `last-read.remark.channel ~)
       |=  [tim=time post=(unit v-post:c)]
       ?&  ?=(^ post)
           !=(author.u.post our.bowl)
       ==
-    =/  unread-id=(unit id-post:c)
-      ?~  unreads  ~
+    =/  count  (lent unreads)
+    =/  unread=(unit [id-post:c @ud])
       ::TODO  in the ~ case, we could traverse further up, to better handle
       ::      cases where the most recent message was deleted.
-      (some -:(rear unreads))
-    =/  count  (lent unreads)
+      ?~  unreads  ~
+      (some -:(rear unreads) count)
     ::  now do the same for all unread threads
     ::
-    =/  [sum=@ud threads=(map id-post:c id-reply:c)]
+    =/  [sum=@ud threads=(map id-post:c [id-reply:c @ud])]
       %+  roll  ~(tap in unread-threads.remark.channel)
-      |=  [id=id-post:c sum=@ud threads=(map id-post:c id-reply:c)]
+      |=  [id=id-post:c sum=@ud threads=(map id-post:c [id-reply:c @ud])]
       =/  parent    (get:on-v-posts:c posts.channel id)
       ?~  parent    [sum threads]
       ?~  u.parent  [sum threads]
       =/  unreads
         %+  skim
-          %~  tap  by
+          %-  bap:on-v-replies:c
           (lot:on-v-replies:c replies.u.u.parent `last-read.remark.channel ~)
         |=  [tim=time reply=(unit v-reply:c)]
         ?&  ?=(^ reply)
             !=(author.u.reply our.bowl)
         ==
-      :-  (add sum (lent unreads))
+      =/  count=@ud  (lent unreads)
```

Comment:
NB: I don't like reusing the name here but it was already doing that for `unreads`.